corehq/apps/callcenter/sync_usercase.py

```diff
@@ -226,31 +229,36 @@ def _call_center_location_owner(user, ancestor_level):
     return owner_id
 
 
-def sync_usercase(user):
+def sync_usercase(user, domain):
     with CriticalSection(get_sync_lock_key(user._id)):
```

Comment:
Should this lock be user ID and domain?  I think as long as this code doesn't modify the user itself, that should be fine.  A task will be queued for each domain the user is a member of, and they'll all be fighting for the same lock.  I imagine we'd want to either let them all run in parallel (by adding domain to the lock key), or run them serially in a single task.

Reply:
@dominicochoa-dimagi I think this still needs to be addressed

Reply:
would this work: 
```python
def get_sync_lock_key(user_id, domain):
    return [f"sync_user_case_for_{user_id}_{domain}"]
```

then:

```python
def sync_usercase(user, domain):
    with CriticalSection(get_sync_lock_key(user._id, domain)):
        domain_obj = Domain.get_by_name(domain)
        _UserCaseHelper.commit(list(_iter_sync_usercase_helpers(user, domain_obj)))
```